tests/unit/testsuite/Bolt/Boltpay/Helper/GraphQLTraitTest.php

```diff
@@ -0,0 +1,100 @@
+<?php
+require_once('TestHelper.php');
+
+class Bolt_Boltpay_Helper_GraphQLTest extends PHPUnit_Framework_TestCase
+{
+    /**
+     * @var PHPUnit_Framework_MockObject_MockObject|Bolt_Boltpay_Helper_GraphQLTest Mocked instance of trait tested
+     */
+    private $currentMock;
+
+    /**
+     * @var PHPUnit_Framework_MockObject_MockObject|Boltpay_Guzzle_ApiClient
+     */
+    private $guzzleClientMock;
+
+    /**
+     * @var PHPUnit_Framework_MockObject_MockObject|\Psr\Http\Message\ResponseInterface
+     */
+    private $responseMock;
+
+    /**
+     * Configure test dependencies, called before each test
+     */
+    public function setUp()
+    {
+        Mage::app('default');
+        $this->currentMock = $this->getMockBuilder('Bolt_Boltpay_Helper_GraphQLTrait')
+            ->disableOriginalConstructor()
+            ->disableOriginalClone()
+            ->disableArgumentCloning()
+            ->setMethods(array('getContextInfo', 'addMetaData', 'getBoltPluginVersion', 'getApiUrl', 'notifyException', 'logException'))
+            ->getMockForTrait();
+        $this->currentMock->method('getBoltPluginVersion')->willReturn('2.3.0');
+        $this->currentMock->method('getApiUrl')->willReturn('https://api.bolt.com/');
+
+
+        $this->guzzleClientMock = $this->getMockBuilder('Boltpay_Guzzle_ApiClient')
+            ->disableOriginalConstructor()
+            ->setMethods(array('post'))
+            ->getMock();
+
+        $this->responseMock = $this->getMockForAbstractClass(
+            '\Psr\Http\Message\ResponseInterface',
+            [],
+            '',
+            false,
+            true,
+            true,
+            ['getBody']
+        );
+
+        Bolt_Boltpay_TestHelper::setNonPublicProperty(
+            $this->currentMock,
+            'apiClient',
+            $this->guzzleClientMock
+        );
```

Comment:
This is ok, but you can do the same thing with

`$this->currentMock->method('getApiClient')->willReturn($this->guzzleClientMock);`

For me, the above is more clear/less obscure.

Reply:
Done